src/docfx/lib/ProcessUtility.cs

```diff
@@ -114,32 +118,37 @@ public static async Task<bool> ReleaseSharedLock(string lockName, string acquire
         /// Acquire a exclusive lock for input lock name
         /// The returned `acquirer` are used for tracking the acquired lock, instead of thread info, since the thread info may change in asynchronous programming model
         /// </summary>
-        public static async Task<(bool acquired, string acquirer)> AcquireExclusiveLock(string lockName)
+        public static Task<(bool acquired, string acquirer)> AcquireExclusiveLock(string lockName, bool retryUntilSuccess = false)
         {
             Debug.Assert(!string.IsNullOrEmpty(lockName));
 
-            var acquired = false;
-            var acquirer = (string)null;
             var lockPath = GetLockFilePath(lockName);
 
-            await ReadAndWriteFile<LockInfo>(lockPath, lockInfo =>
-            {
-                lockInfo = lockInfo ?? new LockInfo();
-                lockInfo = FilterExpiredAcquirers(lockInfo);
+            return retryUntilSuccess ? AcquireWithRetry(lockName, AcquireLock) : AcquireLock();
 
-                if (lockInfo.Type != LockType.None)
+            async Task<(bool, string)> AcquireLock()
+            {
+                var acquired = false;
+                var acquirer = (string)null;
+                await ReadAndWriteFile<LockInfo>(lockPath, lockInfo =>
                 {
-                    return Task.FromResult(lockInfo);
-                }
+                    lockInfo = lockInfo ?? new LockInfo();
+                    lockInfo = FilterExpiredAcquirers(lockInfo);
 
-                acquirer = Guid.NewGuid().ToString();
-                acquired = true;
-                lockInfo.Type = LockType.Exclusive;
-                lockInfo.AcquiredBy = new List<Acquirer> { new Acquirer { Id = acquirer, Date = DateTime.UtcNow } };
-                return Task.FromResult(lockInfo);
-            });
+                    if (lockInfo.Type != LockType.None)
+                    {
+                        return Task.FromResult(lockInfo);
+                    }
+
+                    acquirer = Guid.NewGuid().ToString();
+                    acquired = true;
+                    lockInfo.Type = LockType.Exclusive;
+                    lockInfo.AcquiredBy = new List<Acquirer> { new Acquirer { Id = acquirer, Date = DateTime.UtcNow } };
```

Comment:
Is `lockInfo.AcquiredBy.Add()` enough here instead of allocating a new list?
And possible to consolidate `AcquireLock` with the one in `AcquireSharedLock`?

Reply:
I would like to keep them splitted, the shared logic and different logic are half and half